### DISCUSSION TITLE
ChannelFinder Project

### DISCUSSION THREAD

**Discussion by geniejhang** [ControlSystemStudio/phoebus#1816] [General]
- Phoebus application for generating reports
- Implement NOT search functionality
- A probe like application “Channel Info” which can be launched from the context menu

**Comment by geniejhang** [comment]
@shroffk 
I should've attended the meeting this morning about this matter.
Let me start the discussion from here.

- A probe like application “Channel Info” which can be launched from the context menu

I see this entry already implemented long ago by @shroffk.
Is it different from what you did earlier?
https://github.com/ControlSystemStudio/phoebus/issues/1169

**Reply by geniejhang** [reply]
![screenshot](https://user-images.githubusercontent.com/5515875/110505336-0aee3880-80cc-11eb-9791-8e95e3d0077f.png)

**Reply by shroffk** [reply]
Hi,

The idea is to have the same functionality in other applications... like probe or pvtree... where PV selections have a context menu with the "channel info" which searches channelfinder for that one specific pv.

**Reply by geniejhang** [reply]
Ahhhh. Okay. Now I got what you mean.

**Reply by geniejhang** [reply]
Does it need to be a separate application?

I just replaced supportedTypes from `Channel.class` to `ProcessVariable.class` and then, most of the application using `ProcessVariable` should show Channel Info context menu item.

Slightly modified the UI to display window model dialog instead of `Alert`.

So, like below you can access the context menu:
![probe](https://user-images.githubusercontent.com/5515875/110863638-f908c300-828e-11eb-9f68-8ddabc797e39.png)

and the resulting window is resizable:
![window](https://user-images.githubusercontent.com/5515875/110863668-01f99480-828f-11eb-93bc-508168ffd7d6.png)

I think this is enough for simply inspecting PVs.
One improvement we can make is to put a TextField above the TreeTableView so that user can search PVs interactively with `PVAutocompleteMenu`. 

What do you think?

**Reply by shroffk** [reply]
I think there are a lot of topics here...
Could you join the code-a-thon?

**Reply by geniejhang** [reply]
Pull request submitted. #1826

**Comment by geniejhang** [comment]
- Implement NOT search functionality

This applies for searching with name, owner and properties, right?

**Reply by shroffk** [reply]
I think the first implementation should be for properties and tags

i.e. 
`*pv*name test_property=abc`
return pvname with test_property=abc

`*pv*name test_property=!abc`
return pvname with test_property!=abc

I am still not 100% sure which syntax to use.... is it better to use test_property!=abc

**Reply by geniejhang** [reply]
I would vote for "!="

**Reply by geniejhang** [reply]
Pull request submitted #1829

**Reply by geniejhang** [reply]
Should be merged with https://github.com/ChannelFinder/ChannelFinder-SpringBoot/pull/29
